head_test.go

```diff
@@ -148,16 +127,24 @@ func TestNonDuplicateNaNDatapointsCausesAmendError(t *testing.T) {
 
 	app := hb.Appender()
 	_, err = app.Add(labels.Labels{}, 0, math.Float64frombits(0x7ff0000000000001))
-	if err != nil {
-		t.Fatalf("Failed to add sample: %s", err)
-	}
-	if err = app.Commit(); err != nil {
-		t.Fatalf("Unexpected error committing appender: %s", err)
-	}
+	require.NoError(t, err)
+	err = app.Commit()
+	require.NoError(t, err)
 
 	app = hb.Appender()
 	_, err = app.Add(labels.Labels{}, 0, math.Float64frombits(0x7ff0000000000002))
-	if err != ErrAmendSample {
-		t.Fatalf("Expected error amending sample, got: %s", err)
-	}
+	require.Equal(t, ErrAmendSample, err)
+
+	app = hb.Appender()
+	_, err = app.Add(labels.Labels{}, 0, math.NaN())
+	require.Equal(t, ErrAmendSample, err)
+
+	_, err = app.Add(labels.Labels{}, 1, math.NaN())
+	require.NoError(t, err)
+	err = app.Commit()
+	require.NoError(t, err)
+
+	app = hb.Appender()
+	_, err = app.Add(labels.Labels{}, 1, math.NaN())
+	require.Equal(t, ErrAmendSample, err)
```

Comment:
This shouldn't cause an amend error (it doesn't currently), so it belongs in a differently named test.